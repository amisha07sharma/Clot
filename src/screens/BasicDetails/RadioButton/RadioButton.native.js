import React from "react";
import { TouchableOpacity } from "react-native";
import PropTypes from "prop-types";
import { Wrapper, TextWrapper } from "./RadioButton.style";

const RadioButton = ({ title, isActive = true, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress} testID="Press">
      <Wrapper isActive={isActive}>
        <TextWrapper isActive={isActive}>{title}</TextWrapper>
      </Wrapper>
    </TouchableOpacity>
  );
};

RadioButton.prototype = {
  title: PropTypes.string.isRequired,
  isActive: PropTypes.bool,
  onPress: PropTypes.func.isRequired,
};
export default RadioButton;
