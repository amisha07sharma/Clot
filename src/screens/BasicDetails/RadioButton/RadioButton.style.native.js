import styled from "styled-components/native";
import { Text } from "react-native";
import { BLACK, WHITE, LIGHT_GREY, PURPLE } from "../../../constants/colors";

export const Wrapper = styled.View`
  background-color: ${({ isActive }) => (isActive ? PURPLE : LIGHT_GREY)};
  height: 45px;
  width: 160px;
  border-radius: 100px;
  justify-content: center;
`;

export const TextWrapper = styled(Text)`
  font-weight: 500;
  line-height: 26.73px;
  letter-spacing: -0.4957527816295624px;
  text-align: center;
  color: ${({ isActive }) => (isActive ? WHITE : BLACK)};
  font-size: 16px;
`;
