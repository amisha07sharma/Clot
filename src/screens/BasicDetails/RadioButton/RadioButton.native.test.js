import React from "react";
import { render, fireEvent, screen } from "@testing-library/react-native";
import RadioButton from "./RadioButton";

const mockOnPress = jest.fn();
const mockProps = {
  title: "Title",
  onPress: mockOnPress,
};

const renderComponent = (props) => {
  render(<RadioButton {...mockProps} {...props} />);
};
describe("RadioButton component", () => {
  it("renders correctly with default props", () => {
    const wrapper = renderComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly when isActive is true", () => {
    const wrapper = renderComponent({ isActive: true });
    expect(wrapper).toMatchSnapshot();
  });

  it("handles onPress event correctly", () => {
    renderComponent();
    fireEvent(screen.getByTestId("Press"), "onPress");
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });
});
