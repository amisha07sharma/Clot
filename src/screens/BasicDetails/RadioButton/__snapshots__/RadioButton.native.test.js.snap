// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component renders correctly when isActive is true 1`] = `undefined`;

exports[`RadioButton component renders correctly with default props 1`] = `undefined`;
